### ISSUE TITLE
Nuitka load too many `.so` or `.dll` when building pyqt5 application in `standaone` or `onefile` mode

### ISSUE THREAD

**Issue by r00t1900** [Nuitka/Nuitka#1639]
## version
- nuitka: 0.9.0
- python: 3.8
- os: ubuntu/windows

## problem
I use nuitka to compile many of my pyqt5 applications but I found that every application has almost the same size(linux is about 174M and windows is about 74M). When looking into the `build.dist` folder, I found `libQt5Qml`,`libQt5WebSockets` files which I've never used them. Here is my nuitka command:
```bash
nuitka --output-dir=../build --plugin-enable=pyqt5 --windows-icon-from=res/icon/tray.ico --include-data-dir=res=res --standalone main.py
```
and my python import are like:
```python
from PyQt5 import QtCore, QtGui, QtWidgets
```
even though I import them as `QtCore`, `QtGui` and `QtWidgets`, I use only very few components of them by:
```python
QtWidgets.QSplashScreen(...)
...
```

## expected

Find a correct way to compile my pyqt5 application with the minimum standalone file size(without compress) by adding only necessary libraries. How to get this done, is any thing I can do to improve my nuitka command?

**Comment by kayhayen** [comment]
From a standalone distribution you can remove files right now, that doesn't help though, if you want to use onefile, and plugins can implement DLL inhibition themselves, but that is a bit clumsy, but basically you can overload this:

```
    def removeDllDependencies(self, dll_filename, dll_filenames):
```

But I think, this one is ugly to use, and will likely go away. I used it ages ago to limit Qt plugins from including all of Linux desktops, but it might still be working, but it's not obvious how to use it.

Right now the simplest way is to remove DLLs you do not want to include from the Python installation. Then you can also test if it really is not needed with using standard Python to run your program. Ugly, but workable. Same goes for QML content.

Options to inhibit DLLs are going to come for 1.0 or 1.1, not sure yet, but the preconditions have been implemented, and it's kind of on the roadmap. For a long time plugins copied stuff directly, and there was no good way to control them, but the last ones got revamped, so this is now on. Similar has been done with data files before.

Also make sure to use as little Qt plugins as possible, that makes a large difference too.

**Comment by r00t1900** [comment]
> From a standalone distribution you can remove files right now, that doesn't help though, if you want to use onefile, and plugins can implement DLL inhibition themselves, but that is a bit clumsy, but basically you can overload this:
> 
> ```
>     def removeDllDependencies(self, dll_filename, dll_filenames):
> ```
> 
> But I think, this one is ugly to use, and will likely go away. I used it ages ago to limit Qt plugins from including all of Linux desktops, but it might still be working, but it's not obvious how to use it.
> 
> Right now the simplest way is to remove DLLs you do not want to include from the Python installation. Then you can also test if it really is not needed with using standard Python to run your program. Ugly, but workable. Same goes for QML content.
> 
> Options to inhibit DLLs are going to come for 1.0 or 1.1, not sure yet, but the preconditions have been implemented, and it's kind of on the roadmap. For a long time plugins copied stuff directly, and there was no good way to control them, but the last ones got revamped, so this is now on. Similar has been done with data files before.
> 
> Also make sure to use as little Qt plugins as possible, that makes a large difference too.

Oh yeap, that is really a tough job according to your reply. Looking forward to the newer version :)

**Comment by kayhayen** [comment]
The `--noinclude-dlls` was implemented for 1.0 last minute as is part of the release.

**Comment by kayhayen** [comment]
This is of course resolved by the ability to trim it as of 1.0 already.
